### ISSUE TITLE
Having the ability of setting AppendXForwardedHost to false with IstioIntegration enabled 

### ISSUE THREAD

**Issue by edubonifs** [solo-io/gloo#8855]
### Gloo Edge Product

Enterprise

### Gloo Edge Version

1.15

### Is your feature request related to a problem? Please describe.

We have a client which have several applications considering the X-Fowarded-Host header.

Once they enabled istio Integration, the calls to this app started failing because it seems that, when you enable istio integration with GE, AppendXForwardedHost is enforced to true: https://github.com/solo-io/gloo/blob/114838123221fb71bd27af1d26f878dc483ecf4b/projects/gloo/pkg/plugins/istio_integration/plugin.go#L54

I would like to know why is this setting enforced to true and if we could modify this behaviour

**Comment by gsustek** [comment]
latest update:

> @kdorosh
Sounds like a product request for an RFE so we can handle the istio stuff for you via the gloo VS abstraction. 

solution for https://istio.io/latest/docs/reference/config/networking/virtual-service/#HTTPRewrite

the best approach, for us, would be to have ability to [enable/disable rewrite with istiointegratio enabled](https://github.com/solo-io/gloo/issues/8855)
this solution what you propose is more per gloo virtualservice.
what exactly authority string value should be for not to rewrite eg, to just rollout it back to user-agent authority value?

**Comment by edubonifs** [comment]
Hi, 
I think some of our customers have a huge number of RouteTables and matches in VirtualServices, so I think the best approach would be to make a global option for setting AppendXForwardedHost to false instead of using HostRewrite or HttpRewrite, as customer would need to go through every matcher or every RouteTable.
If we could have a global option for AppendXForwardedHost we would avoid this for customers with high number of resources

**Comment by gsustek** [comment]
latest update from gloo engineering:

> When Istio integration is enabled there's a sidecar added on the Edge GW proxy to handle mTLS. Since the traffic flows through this sidecar proxy on the GW proxy pod it cannot route traffic to the K8s backend service without seeing the host header of the K8s service which is set by the Edge GW proxy container. Using an EnvoyFilter to set the authority on the Edge GW proxy sidecar post routing back to the client host value is the only viable solution here.

this would be 

```
apiVersion: networking.istio.io/v1alpha3
kind: EnvoyFilter
metadata:
  name: host-rewrite
  namespace: application-namespace
spec:
  configPatches:
  - applyTo: HTTP_ROUTE
    match:
      context: SIDECAR_OUTBOUND
    patch:
      operation: MERGE
      value:
        route:
          host_rewrite_header: x-forwarded-host
  workloadSelector:
    labels:
      gloo: gateway-proxy
```
      
     RH Service Mesh [does not allow/restrict](https://access.redhat.com/articles/6972785) Envoy filters...

**Comment by edubonifs** [comment]
Hi @nrjpoddar , pinging due to the slack conversation, I understand it is mandatory to rewrite the Host header in order to know the K8s service which is set by the Edge GW proxy container.
But is it also mandatory to set routeAction.Route.AppendXForwardedHost = true ?
Checking the code:
```
func (p *plugin) ProcessRoute(params plugins.RouteParams, in *v1.Route, out *envoy_config_route_v3.Route) error {
	//check if this is a kubernetes destination
	dest, ok := in.GetRouteAction().GetDestination().(*v1.RouteAction_Single)
	if !ok {
		// With current use cases it only makes sense to handle k8s single destinations
		// If this use case expands we would need to figure out Istio's expected hostname format for each type of destination.
		return nil
	}
	hostInMesh, err := GetHostFromDestination(dest, params.Snapshot.Upstreams)
	if err != nil {
		return err
	}
	if hostInMesh == "" {
		return nil
	}
	routeAction, ok := out.GetAction().(*envoy_config_route_v3.Route_Route)
	if !ok || routeAction.Route.GetHostRewriteSpecifier() != nil {
		// This only applies to routes that do not already have a host rewrite
		return nil
	}
	//Set the host rewrite and x-forwarded-host header
	routeAction.Route.HostRewriteSpecifier = &envoy_config_route_v3.RouteAction_HostRewriteLiteral{
		HostRewriteLiteral: hostInMesh,
	}
	routeAction.Route.AppendXForwardedHost = true
	return nil
}
```

we set the hostRewrite and set AppendXForwardedHost in two different steps, is appending the header to X-Forwarded-Host also mandatory to get the routing to the istio Upstream service working?

**Comment by edubonifs** [comment]
Hi, 
Adding more information.
For the new istioIntegration to work, we need to specify these values:
```
global:
  istioIntegration:
    enableIstioSidecarOnGateway: true
    disableAutoinjection: true
  istioSDS:
    enabled: true
  glooMtls:
    istioProxy:
      image:
        registry: docker.io/istio
        repository: proxyv2
        tag: 1.18.2
    sds:
      image:
        repository: sds
        tag: 1.15.7
```

If you go to the [code](https://github.com/solo-io/gloo/blob/114838123221fb71bd27af1d26f878dc483ecf4b/projects/gloo/pkg/plugins/istio_integration/plugin.go#L54), it says:
```
When istio integration is enabled, we need to access k8s services using a host that istio will recognize (servicename.namespace)
We do this by adding a hostRewrite for kube destinations kube upstreams. In case the upstream also wants the original host,
we also set x-forwarded-host
```

So every time we have the sidecar injected, we enter into this loop.

However, with the new istio integration, this header AppendXForwardedHost is optional now, as it was used for the istio-proxy to know wether to use mtls or not.

With the new istio integration, we are now using glooctl istio enable-mtls, which is explained in this [doc](https://docs.solo.io/gloo-edge/latest/reference/cli/glooctl_istio_enable-mtls/)

With this setting, we specify on the Upstream wether the istio-proxy will use mtls or not, so AppendXForwardedHost should be no more needed

**Comment by gsustek** [comment]
@SantoDE  any movement on this topic? We are expecting this toogle 💯   🙏

**Comment by nrjpoddar** [comment]
cc @DuncanDoyle @kcbabo

**Comment by bewebi** [comment]
This feature has been implemented and released in OSS v1.17.0-beta7
It will also be released in OSS v1.16.4 and v1.15.23 and EE v1.17.0-beta1, v1.16.2, and v1.15.13

To opt out of appending the x-forwarded-host header as part of the Istio integration users may set the `global.istioIntegration.AppendXForwardedHost` Helm value to `false`
The value defaults to `true`, preserving the existing behavior
